mlflow/utils/proto_json_utils.py

```diff
@@ -189,4 +194,16 @@ def parse_tf_serving_input(inp_dict):
                 "Failed to parse data as TF serving input. The length of values for"
                 " each input/column name are not the same"
             )
+
+    # Attempt casting
+    if schema is not None:
```

Comment:
can we cast right when we create the array - i.e. call np.array(data, dtype=...))? I am worried that casting at the end adds another data copy.